src/pushover_complete/pushover_api.py

```diff
@@ -136,10 +140,14 @@ def _send_message(self, user, message, device=None, title=None, url=None, url_ti
             'sound': sound,
             'html': html
         }
-
+        
+        if image_path is not None and os.path.isfile(image_path):
+            with open(image_path, 'rb') as f:
+                file = {'attachment': (image_path, f)}
+                return self._generic_post('messages.json', payload=payload, session=session, files=file)
```

Comment:
I really hate the need for an if here but I can't see a more elegant way to do it.

Reply:
I could drop the conditional and just rely on exception handling, but that would still be:
```python
try:
    with open( image_path, 'rb') as f :
        file = {'attachment': f}
        return self._generic_post('messages.json', payload=payload, session=session, files=file)
except IOError:
    pass
return self._generic_post('messages.json', payload=payload, session=session)
```